src/dumper.rs

```diff
@@ -431,6 +431,17 @@ impl<'a> Dumper<'a> {
             .map(AsRef::as_ref)
             .ok_or_else(|| anyhow!("could not resolve type {}", name.as_str()))
     }
+
+    pub fn report(&self) -> impl fmt::Display {
```

Comment:
```suggestion
    pub fn report(&self) -> String {
```
no need to return an opaque type here